src/main/java/net/dv8tion/jda/api/interactions/components/selections/EntitySelectMenu.java

```diff
@@ -0,0 +1,469 @@
+         */
+        @Nonnull
+        public Builder setRequiredRange(int min, int max)
+        {
+            Checks.check(min <= max, "Min Values should be less than or equal to Max Values! Provided: [%d, %d]", min, max);
+            return setMinValues(min).setMaxValues(max);
+        }
+
+        /**
+         * Configure whether this select menu should be disabled.
+         * <br>Default: {@code false}
+         *
+         * @param  disabled
+         *         Whether this menu is disabled
+         *
+         * @return The same builder instance for chaining
+         */
+        @Nonnull
+        public Builder setDisabled(boolean disabled)
+        {
+            this.disabled = disabled;
+            return this;
+        }
+
+        /**
+         * The custom id used to identify the select menu.
+         *
+         * @return The custom id
+         */
+        @Nonnull
+        public String getId()
+        {
+            return customId;
+        }
+
+        /**
+         * The type(s) used in the select menu.
+         *
+         * @return The type(s)
+         */
+        @Nonnull
+        public EnumSet<SelectType> getType() { return type; }
+
+        /**
+         * The allowed channels used in the select menu.
+         *
+         * @return The allowed channels
+         */
+        @Nonnull
+        public EnumSet<ChannelType> getChannelFilters() { return channelFilters; }
+
+        /**
+         * Placeholder which is displayed when no selections have been made yet.
+         *
+         * @return The placeholder or null
+         */
+        @Nullable
+        public String getPlaceholder()
+        {
+            return placeholder;
+        }
+
+        /**
+         * The minimum amount of values a user has to select.
+         *
+         * @return The min values
+         */
+        public int getMinValues()
+        {
+            return minValues;
+        }
+
+        /**
+         * The maximum amount of values a user can select at once.
+         *
+         * @return The max values
+         */
+        public int getMaxValues()
+        {
+            return maxValues;
+        }
+
+        /**
+         * Whether the menu is disabled
+         *
+         * @return True if this menu is disabled
+         */
+        public boolean isDisabled()
+        {
+            return disabled;
+        }
+
+        /**
+         * Creates a new {@link EntitySelectMenu} instance if all requirements are satisfied.
+         *
+         * @return The new {@link EntitySelectMenu} instance
+         */
+        @Nonnull
+        public EntitySelectMenu build()
+        {
+            Checks.check(minValues <= maxValues, "Min values cannot be greater than max values!");
+
+            int typeCode;
+
+            if (type.size() == 1)
+                typeCode = type.iterator().next().getKey();
+            else if (type.size() == 3 && type.contains(SelectType.ROLE) && type.contains(SelectType.USER))
```

Comment:
Should this be `type.size() == 2`? Setting `SelectType.ROLE` and `SelectType.USER` causes this to throw an IllegalArgumentException.

Reply:
Yep, thanks for pointing that out. I remembered that the mentionable select menu doesn't include channels about half way through doing this haha.